src/fswAlgorithms/orbitControl/lambertSolver/lambertSolver.cpp

```diff
@@ -0,0 +1,669 @@
+    }
+    else if (this->numberOfRevolutions == 0){
+        // if zero orbits are completed, only one solution exists.
+        // Only populate information for 1st solution, 2nd solution remains zero message payload
+        eigenVector3d2CArray(this->vvecs.at(0), lambertSolutionOutMsgBuffer.v1);
+        eigenVector3d2CArray(this->vvecs.at(1), lambertSolutionOutMsgBuffer.v2);
+        lambertSolutionOutMsgBuffer.valid = 1; // solution 1 is valid
+
+        lambertPerformanceOutMsgBuffer.x = this->X;
+        lambertPerformanceOutMsgBuffer.numIter = this->numIter;
+        lambertPerformanceOutMsgBuffer.errX = this->errX;
+    }
+    else{
+        // if one or more orbits are completed, and the requested time of flight is long enough, two solutions exist
+        eigenVector3d2CArray(this->vvecs.at(0), lambertSolutionOutMsgBuffer.v1);
+        eigenVector3d2CArray(this->vvecs.at(1), lambertSolutionOutMsgBuffer.v2);
+        lambertSolutionOutMsgBuffer.valid = 1; // solution 1 is valid
+        eigenVector3d2CArray(this->vvecsSol2.at(0), lambertSolutionOutMsgBuffer.v1Sol2);
+        eigenVector3d2CArray(this->vvecsSol2.at(1), lambertSolutionOutMsgBuffer.v2Sol2);
+        lambertSolutionOutMsgBuffer.validSol2 = 1; // solution 2 is valid
+
+        lambertPerformanceOutMsgBuffer.x = this->X;
+        lambertPerformanceOutMsgBuffer.numIter = this->numIter;
+        lambertPerformanceOutMsgBuffer.errX = this->errX;
+        lambertPerformanceOutMsgBuffer.xSol2 = this->XSol2;
+        lambertPerformanceOutMsgBuffer.numIterSol2 = this->numIterSol2;
+        lambertPerformanceOutMsgBuffer.errXSol2 = this->errXSol2;
+    }
+
+    // Write to the output messages
+    this->lambertSolutionOutMsg.write(&lambertSolutionOutMsgBuffer, this->moduleID, currentSimNanos);
+    this->lambertPerformanceOutMsg.write(&lambertPerformanceOutMsgBuffer, this->moduleID, currentSimNanos);
+}
+
+/*! This method computes the problem geometry for the given parameters of Lambert's problem.
+    The orbit frame is also determined.
+    @return void
+*/
+void LambertSolver::problemGeometry()
+{
+    double c = (this->r2vec - this->r1vec).norm(); // chord length
+    double r1 = this->r1vec.norm();
+    double r2 = this->r2vec.norm();
+    double s = 1.0/2.0*(r1+r2+c); // semiperimeter
+
+    // lambda parameter
+    this->lambda = sqrt(1.0-c/s);
+    // non-dimensional time-of-flight
+    this->TOF = sqrt(2.0*this->mu/pow(s,3))*this->transferTime;
+
+    // compute orbit frame unit vectors
+
+    // radial direction at for initial and final position vector
+    Eigen::Vector3d i_r1 = this->r1vec/r1;
+    Eigen::Vector3d i_r2 = this->r2vec/r2;
+    // check alignment of the two position vectors
+    double sinTheta = i_r1.cross(i_r2).norm();
+    if (abs(sinTheta) < sin(this->alignmentThreshold * M_PI/180.)){
+        bskLogger.bskLog(BSK_WARNING,
+                         "lambertSolver: position vectors r1 and r2 are too aligned, that is, the angle between "
```

Comment:
The unit test throws this warning message when running `purest`.  Is this expected?

Reply:
Yes, the unit test specifically tests for the case where the position vectors are very aligned (transfer angle of 0, 180, 360, ... degrees). In this case, this warning should be thrown.